### PR TITLE
Audit API for status timeline

### PR DESCRIPTION
**Pull request by Tusharmahajan12**
created new table audit_logs in dev and taken old and new changes done

### CODE REVIEW

File: src/src/user/user.controller.ts [hunk 1]
```diff
@@ -162,4 +162,19 @@ export class UserController {
 	) {
 		return this.userService.getAadhaarDetails(id, response);
 	}
+
+	@Get('/audit/:conext_id/:context')
```
Comment (manojLondhe) [inline src/src/user/user.controller.ts:166]: `conext_id` => `context_id` change everywhere

File: src/src/beneficiaries/beneficiaries.service.ts [hunk 2]
```diff
@@ -610,10 +611,11 @@ export class BeneficiariesService {
 		// return this.hasuraService.delete(this.table, { id: +id });
 	}
 
-	public async statusUpdate(req: any) {
+	public async statusUpdate(req: any, header: any) {
```
Comment (manojLondhe) [inline src/src/beneficiaries/beneficiaries.service.ts:614]: from controller you are passing body, request
where is header coming from?

File: src/src/user/user.controller.ts (hunk 1 above)
Comment (manojLondhe) [reply src/src/user/user.controller.ts]: let the endpoint be as

/api/user/audit/:context/:context_id

File: src/src/user/user.service.ts [hunk 3]
```diff
@@ -1048,4 +1049,92 @@ export class UserService {
...
+					'context_id',
+					'updated_at',
+					'created_at',
+					'updated_by_user',
+				],
+			);
+			return res;
+		}
+	}
+	public async getAuditDetails(context_id, context, req: any, resp: any) {
```
Comment (manojLondhe) [inline src/src/user/user.service.ts:1101]: add blank at the start of new method

File: src/src/user/user.controller.ts [hunk 4]
```diff
@@ -162,4 +162,19 @@ export class UserController {
...
+	getAuditDetails(
```
Comment (manojLondhe) [inline src/src/user/user.controller.ts:167]: rename from
getAuditDetails
to
getAuditLogs everywhere

File: src/src/user/user.service.ts [hunk 5]
```diff
@@ -1048,4 +1049,92 @@ export class UserService {
...
+				success: false,
+				status: 'Not Found',
+				message: 'Audit Logs Not Found',
+				data: {},
+			});
+		} else {
+			return resp.status(200).json({
+				success: true,
+				message: 'Audit Logs found successfully!',
+				data: { result: result },
```
Comment (manojLondhe) [inline src/src/user/user.service.ts:1136]: o/p not as per standard format decided?

File: src/src/user/user.service.ts [hunk 6]
```diff
@@ -1048,4 +1049,92 @@ export class UserService {
 			};
 		}
 	}
+
+	async auditLogs(
```
Comment (manojLondhe) [inline src/src/user/user.service.ts:1053]: rename 
auditLogs =>  addAuditLog

**Review by manojLondhe — CHANGES_REQUESTED** [review]

File: src/src/beneficiaries/beneficiaries.service.ts (hunk 2 above)
Comment (Tusharmahajan12) [reply src/src/beneficiaries/beneficiaries.service.ts]: from controller now we are passing body and request. 
In service we are getting body and request.

**Pull request merged**